src/java_bytecode/java_object_factory.cpp

```diff
@@ -1026,3 +1044,49 @@ void gen_nondet_init(
 
   init_code.append(assignments);
 }
+
+/// Call object_factory with a default (identity) pointer_type_selector
```

Comment:
Sorry I don't understand this comment - the `select_pointer_typet` is the identity selector. I will see if I can make it work with a reference or a `unique_ptr`